### ISSUE TITLE
Provide an option to specify additional pool configuration

### ISSUE THREAD

**Issue by anujsrc** [kul/pool#2]
[GenericObjectPool](http://commons.apache.org/proper/commons-pool/apidocs/org/apache/commons/pool2/impl/GenericObjectPool.html) and [GenericKeyedObjectPool](http://commons.apache.org/proper/commons-pool/apidocs/org/apache/commons/pool2/impl/GenericKeyedObjectPool.html) provide an option to specify [BaseObjectPoolConfig](http://commons.apache.org/proper/commons-pool/apidocs/org/apache/commons/pool2/impl/BaseObjectPoolConfig.html) that can be used for variety of configuration of the pool, like-

`DEFAULT_MIN_EVICTABLE_IDLE_TIME_MILLIS` : The default value for the minEvictableIdleTimeMillis configuration attribute.
`DEFAULT_TIME_BETWEEN_EVICTION_RUNS_MILLIS`: The default value for the timeBetweenEvictionRunsMillis configuration attribute.

It would be good to provide an option explicitly for such configuration or pass it via kwargs.

**Comment by kul** [comment]
Yes, optional kwargs would be prefered method as it keeps api usage simple.
